src/scale-manager/ScaleManager.spec.js

```diff
@@ -42,4 +53,61 @@ describe('Scale Manager', () => {
       done();
     }, 1);
   });
+
+  describe('--Anchor Handling--', () => {
+    beforeEach(() => {
+      sm.entities = [];
+    });
+
+    it('should add an anchor', () => {
+      sm.addEntity(new Anchor());
+
+      expect(sm.entities.length).to.equal(1);
+    });
+
+    it('should not add an anchor if it already exists', () => {
+      const testAnchor = new Anchor();
+
+      sm.addEntity(testAnchor);
+      sm.addEntity(testAnchor);
+
+      expect(sm.entities.length).to.equal(1);
+    });
+
+    it('should remove anchors', () => {
+      const testAnchor = new Anchor();
+
+      sm.addEntity(testAnchor);
+      sm.removeEntity(testAnchor);
+
+      expect(sm.entities.length).to.equal(0);
+    });
+
+    it('should invoke anchor update when added', done => {
+      const testAnchor = new Anchor({
+        callback: () => {
+          done();
+        }
+      });
+      sm.addEntity(testAnchor);
+    });
+
+    it('should invoke anchor update on resize', done => {
+      let count = 0;
+
+      const testAnchor = new Anchor({
+        callback: () => {
+          if (count == 1) {
```

Comment:
Because 2 resize events are being emitted, should count equal 2, as the callback is done when added as well?

Reply:
The first time the callback is invoked count is 0, (when it's added to the scale manager). I believe dispatching the resize event twice is because of a bug in the headless version of firefox? @SmithAF

Reply:
Yes, added: count is 0, first event: count is 1, second event: count is 2. Here you'd be running code after the test completes.